# Test de l’API de Scoring Crédit

## Projet MLOps – Modèle de scoring crédit avec API et déploiement continu

**Auteur : [Ton Nom & Prénom]**  
**Date : [à compléter]**

---

### Introduction

Ce notebook permet de tester l’API de scoring crédit développée et déployée dans le cadre du projet MLOps.  
L’API, hébergée sur Heroku, retourne la probabilité de défaut de paiement ainsi qu’une décision d’octroi ou non de crédit à partir de l’identifiant d’un client.

**Objectifs de ce notebook :**
- Vérifier que l’API répond correctement pour un client existant (retourne la prédiction attendue)
- Vérifier la gestion d’un client inexistant (retourne une erreur appropriée)
- Illustrer l’intégration de l’API dans une démarche de validation MLOps

**URL de l’API testée :**  
[https://credit-scoring-jademayalb-db8bcc609fed.herokuapp.com/predict/{client_id}](https://credit-scoring-jademayalb-db8bcc609fed.herokuapp.com/predict/{client_id})

In [5]:
# Imports
import requests
import pandas as pd

## Test pour un client existant

Nous allons interroger l’API pour l’identifiant client `100091`.

In [6]:
url = 'https://credit-scoring-jademayalb-db8bcc609fed.herokuapp.com/predict/100091'
response = requests.get(url)
print(f"Status code : {response.status_code}")
result = response.json()
print(result)

Status code : 200
{'client_id': 100091, 'decision': 'REFUSÉ', 'model_name': 'LightGBM_Balanced', 'probabilite_defaut': 0.5226794561096456, 'seuil_optimal': 0.52, 'status': 'OK'}


## Test pour un client inexistant

Nous allons interroger l’API pour un identifiant client qui n’existe pas (`999999999`).

In [7]:
url_inexistant = 'https://credit-scoring-jademayalb-db8bcc609fed.herokuapp.com/predict/999999999'
response_inexistant = requests.get(url_inexistant)
print(f"Status code : {response_inexistant.status_code}")
result_inexistant = response_inexistant.json()
print(result_inexistant)

Status code : 404
{'erreur': 'Client ID 999999999 introuvable', 'status': 'NOT_FOUND'}


## 4. Analyse des réponses

- **Client existant :** on doit obtenir un code 200, avec une réponse contenant la probabilité de défaut et la décision.
- **Client inexistant :** on doit obtenir un code 404, avec un statut `"NOT_FOUND"`.

On peut également afficher la réponse dans un DataFrame pour une meilleure lisibilité.

In [9]:
# Pour le client existant
pd.DataFrame([result])

,client_id,decision,model_name,probabilite_defaut,seuil_optimal,status
0,100091,REFUSÉ,LightGBM_Balanced,0.522679,0.52,OK


In [10]:
# Pour le client inexistant
pd.DataFrame([result_inexistant])

,erreur,status
0,Client ID 999999999 introuvable,NOT_FOUND


## 5. Conclusion

Ce notebook démontre que l’API répond de façon appropriée selon que le client existe ou non, et permet de valider son intégration dans le pipeline MLOps.